### ISSUE TITLE
Use AuthSession cookie for querying instead of basic auth in healthcheck

### ISSUE THREAD

**Issue by dianabarsan** [medic/cht-core#8737]
**Describe the issue**
We're discussing increasing [pbkdf2 iterations](https://github.com/medic/cht-core/issues/8338) for CouchDb. 
This will mean that computing the hashed password every time basic auth is used will become a costly server-side operation.  
`haproxy-healthcheck` polls `_membership` with basic auth every 5 seconds. 

**Describe the improvement you'd like**
Instead of using basic auth for every request, make one initial _session request, and use the provided AuthSession cookie to authenticate all following requests within one conf run. 

**Describe alternatives you've considered**
None.

**Comment by fardarter** [comment]
For a healthcheck, would _up not be a better API as per: https://docs.couchdb.org/en/stable/config/auth.html#chttpd/require_valid_user_except_for_up 

If membership did not meet quorum, would you not see couch reject requests?

Trying to understand for our own HAProxy config.

**Comment by dianabarsan** [comment]
Unfortunately the _up command doesn't check for this, it just checks if the server is not in maintenance mode. 
https://docs.couchdb.org/en/stable/api/server/common.html#get--_up

**Comment by fardarter** [comment]
> Unfortunately the _up command doesn't check for this, it just checks if the server is not in maintenance mode. https://docs.couchdb.org/en/stable/api/server/common.html#get--_up

From the doc you linked: `Confirms that the server is up, running, and ready to respond to requests. `

It looks like it responds with 404 if in maintenance. 

If the server did not have quorum on membership, would it respond that it's ready for requests?

I realise this isn't a direct check of membership but would you be able to infer from it whether there was a membership quorum? Ie, would it not respond with something other than 200 or 404 if there was no quorum?

If you think this is worth exploring, I can put this on our map to try out.

**Comment by dianabarsan** [comment]
> If the server did not have quorum on membership, would it respond that it's ready for requests?

Yes, it unfortunately does. Quorum is not what it's thought out to be :)

**Comment by fardarter** [comment]
What a pain.

**Comment by dianabarsan** [comment]
Closing this as the latest versions of CouchDb have optimized basic auth password encoding/decoding, along with password caching.
